dotCMS/src/main/java/com/dotmarketing/common/reindex/ReindexThread.java

```diff
@@ -228,13 +246,52 @@ private void runReindexLoop() {
 
                 }
             } catch (Throwable ex) {
+                // Check if this is a shutdown-related exception
+                if (isShutdownRelated(ex) || ShutdownCoordinator.isRequestDraining()) {
+                    Logger.info(this, "Shutdown-related exception in ReindexThread, stopping operations: " + ex.getMessage());
+                    break;
+                }
                 Logger.error(this, "ReindexThread Exception", ex);
                 ThreadUtils.sleep(SLEEP_ON_ERROR);
             } finally {
                 DbConnectionFactory.closeSilently();
             }
             sleep();
         }
+        
+        // Clean up bulk processor on exit
+        try {
+            if (bulkProcessor != null) {
+                closeBulkProcessor(bulkProcessor);
+            }
+        } catch (Exception e) {
+            Logger.debug(this, "Exception while closing bulk processor during shutdown: " + e.getMessage());
+        }
+    }
+    
+    /**
+     * Check if an exception is related to shutdown (database connection closed, etc.)
+     */
+    private boolean isShutdownRelated(Throwable ex) {
+        if (ex == null) return false;
+        
+        String message = ex.getMessage();
+        if (message != null) {
+            message = message.toLowerCase();
+            // Common shutdown-related error messages
+            if (message.contains("connection") && (message.contains("closed") || message.contains("terminated"))) {
```

Comment:
It would interesting to have such as a Classifier to tell if there is or not an error feeling in the msg

Reply:
Yeah, don't like the specific message.contains myself.   Will look at this again.